code/modules/vehicles/spacepod/actions.dm

```diff
@@ -0,0 +1,119 @@
+/datum/action/vehicle/sealed/kick_out
+	name = "Kick Out"
+	desc = "Kick someone out of this vehicle."
+	button_icon = 'icons/mob/actions/actions_pod.dmi'
+	button_icon_state = "kickout"
+	background_icon_state = "bg_tech"
+	overlay_icon_state = "bg_tech_border"
+
+/datum/action/vehicle/sealed/kick_out/IsAvailable(feedback = FALSE)
+	. = ..()
+	if(!.)
+		return
+	if(HAS_TRAIT(owner, TRAIT_RESTRAINED))
+		if (feedback)
+			owner.balloon_alert(owner, "restrained!")
+		return FALSE
+	return TRUE
+
+/datum/action/vehicle/sealed/kick_out/Trigger(trigger_flags)
+	. = ..()
+	var/list/occupants = vehicle_entered_target.occupants.Copy()
+	occupants -= owner
+
+	if(!length(occupants))
+		vehicle_entered_target.balloon_alert(owner, "nobody else!")
+		return
+	var/mob/living/to_kick = length(occupants) == 1 ? occupants[1] : tgui_input_list(owner, "Kick whom?", "Kick whom?", occupants)
+	if(!to_kick || owner.loc != vehicle_entered_target) //kicked out before them get lost buddy
+		return
+	to_kick.Knockdown(1 SECONDS)
+	vehicle_entered_target.mob_exit(to_kick, randomstep = TRUE)
```

Comment:
can one of the occupants leave while im making my choice on who to kick? if so, ull need to check their location as well so i cant knockdown someone whos already exitted